drivers/bbram/bbram_stm32.c

```diff
@@ -65,6 +66,10 @@ static int bbram_stm32_write(const struct device *dev, size_t offset, size_t siz
 		return -EFAULT;
 	}
 
+#ifdef TAMP
+	LL_PWR_EnableBkUpAccess();
```

Comment:
question : because the TAMP registers are part of the BackUp domain, like the RTC, enabling the LL_PWR_EnableBkUpAccess should already be done in the same way as enabling the backup domain register for RTC
`#if defined(PWR_CR_DBP) || defined(PWR_CR1_DBP) || defined(PWR_DBPR_DBP)`

This is valid for stm32g0 but also stm32g4, stm32h5 and maybe more

Reply:
I can indeed change that. Note however that the stm32g4 and stm32h5 also have the backup register in the TAMP peripheral. My idea was to match the test done at the top of that file on the TAMP peripheral to avoid doing that operation on devices which have the backup registers as part of the RTC peripheral.